### PR TITLE
feat: Adds tag component

### LINKED ISSUE #176: Implement tags
[UXPin (Pattern Library)](https://preview.uxpin.com/5ca5c040146a4ebb4536c10998dd84d1199e5d79#/pages/134488442/specification/sitemap)

cauldron tags pattern

### PR DESCRIPTION
**Pull request by schne324**
resolves #176

---

tag component unit test coverage 100%

### CODE REVIEW

File: packages/styles/variables.css [hunk 1]
```diff
@@ -152,4 +152,9 @@
 
   /* list */
   --list-separator: rgba(153, 153, 153, 0.15);
+
+  /* tags */
+  --tag-text-color: #333333;
```
Comment (schne324) [inline packages/styles/variables.css:157]: we still haven't implemented our full color palette so for now, I'm adding new vars for tags...once we implement our palette I can replace references to these with the palette var names

Comment (stephenmathieson) [reply packages/styles/variables.css:157]: Is there an issue for adding the remaining variables?

File: docs/patterns/components/Tag/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      <div>
+        <Demo
+          component={Tag}
+          states={[
+            {
+              children: 'Value'
+            },
+            {
+              children: (
+                <>
```
Comment (stephenmathieson) [inline docs/patterns/components/Tag/index.js:18]: What's the Fragment for?

**Review by stephenmathieson — APPROVED** [review]

File: docs/patterns/components/Tag/index.js (hunk 2 above)
Comment (schne324) [reply docs/patterns/components/Tag/index.js:18]: there are 2 sibling reactNode's here: 
- `<TagLabel />`
- "value" (string)

File: packages/styles/variables.css (hunk 1 above)
Comment (schne324) [reply packages/styles/variables.css:157]: yes I believe so...there's a draft PR too

File: docs/patterns/components/Tag/index.js (hunk 2 above)
Comment (stephenmathieson) [reply docs/patterns/components/Tag/index.js:18]: Ah yep, I thought it was a child of the `<TagLabel>`. My bad 👍

**Pull request merged**